soundcloud-api/src/main/java/com/jlubecki/soundcloud/webapi/android/models/Activity.java

```diff
@@ -0,0 +1,58 @@
+package com.jlubecki.soundcloud.webapi.android.models;
+
+/**
+ * Created by takahawk on 7/10/16.
+ */
+public class Activity {
+
+    public ActivityOrigin origin;
```

Comment:
At first I thought the same way. But fact that origin can be also playlist makes it very difficult to find a proper solution.
I realize that my implementation isn't perfect but I need this API functions for project i'm working for. I just wanted to share, mb someone also need it. You are free not to merge my pull request of course if you want to implement it your way.